[WIP] Cleanup Cmake builds

amarpad opened the pull request:
## Summary

Update CMake dependencies for Magma builds

## Test Plan

Built Magma
Things still pending
- mme_test, session_manager_test and figuring out how official builds are generated
- validating clion alternate workflow works for the team

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

Commit:
Format cmakelist files consistently

Tab: 4 spaces
Indentation: 2
Continuing Indentation: 4

No-op change, outside just making file formats consistent

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

Commit:
Rebuild common services using a super build

Today we implicitly build the common libraries vs importing them as an
external project from the core and session manager projects.
Create a super build makefile that builds each of the libraries as an
exported package that can be imported by other binaries that depend on it.

This change doesn't change the scope of the compilation. It is likely that
we should have fewer .a's but this change doesn't change the scope of
compilation.

This change also renames the CONFIG module to MAGMA_CONFIG to allow
find_package to find MAGMA_CONFIG vs libconfig

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

Commit:
Create a core superbuild

Create a core super build that depends on common libraries

This prevents the implicit ordering of builds between common and core and
adds an explicit, DEPENDS directive followed by find_packages for importing
exported common library targets.

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

Commit:
Update gitignore

Update gitignore to not ignore lib directory under lte/gateway/c

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

Commit:
Remove an unused function

mme_app_resume_timer is unused

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

Commit:
Fix clearing of PCO attributes to preserve const correctness

Move clearing of PCO to caller to preseve const correctness of
attribtes


Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

Commit:
Use superbuilds for session_manager

Create a super build for session manager to allow
for explicit dependency on magma common

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

rdefosse commented:
@amarpad 

as mentioned in DM on Slack, the build is made this way:

```
export FEATURES=mme_oai
export MAGMA_ROOT=/magma
export BUILD_TYPE=RelWithDebInfo
export C_BUILD=/build/c

cd $MAGMA_ROOT/lte/gateway
echo $FEATURE
make build_oai
make build_sctpd
```

The common part looks good:

```
cd  /build/c/magma_common && awk '{if (/^CMAKE_EXPORT_COMPILE_COMMANDS/) gsub(/OFF/, "ON"); print}' CMakeCache.txt > tmp && mv tmp CMakeCache.txt
ninja -C  /build/c/magma_common
...
[64/64] Completed 'PolicyDb'
```
The oai part does not start at all.

```
mkdir -p  /build/c/oai
cd  /build/c/oai &&  cmake  /magma/lte/gateway/c/oai -DCMAKE_BUILD_TYPE=RelWithDebInfo  -DS6A_OVER_GRPC=False -DCMAKE_C_FLAGS="-Wall " -DCMAKE_CXX_FLAGS="-Wall " -GNinja
CMake Error: The source directory "/magma/lte/gateway/c/oai" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
make: *** [build_oai] Error 1
Makefile:152: recipe for target 'build_oai' failed
The command '/bin/sh -c cd $MAGMA_ROOT/lte/gateway &&     echo $FEATURES &&     make build_oai &&     make build_sctpd' returned a non-zero code: 2
MAGMA-OAI-MME DOCKER IMAGE BUILD: KO
~

Commit:
Fix const correctness for sgw_s8_send_create_session_response arg

sgw_s8_send_create_session_response mutates the pco options struct
so remove the const qualifier for the session_rsp pointer variable.

Fixes a compiler warning.

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

Commit:
Adding missing header function definitions

Add missing header definitions for sgw_s8_s11 handlers

Fixes compile

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

Commit:
Makefile updates for the new path

Trivial update to point at the superbuilds

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

Commit:
Create superbuild for sctpd

Add a superbuild for sctpd.
Also fix some of the include directories

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

Commit:
Add superbuild for connection tracker

Add a super build for Connection Tracker which depends on common

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

themarwhal commented on lte/gateway/c/core/oai/CMakeLists.txt:
```diff
@@ -58,19 +58,6 @@ set(CMAKE_EXE_LINKER_FLAGS_DEBUG  "${CMAKE_EXE_LINKER_FLAGS_DEBUG} -fprofile-arc
 
-################################################################
-# Add a flag to enable sentry if the library exists
-################################################################
```
Can we keep this? (Or did it move to some other file?)

amarpad replied:
rebase issue I think. Let me take a pass at this file again.

themarwhal commented:
So is the broad idea to do the following? 
* put all src files into [directory/src/](url)
* put any header files that need to be used somewhere else in `directory/includes/`

themarwhal commented on lte/gateway/c/core/oai/CMakeLists.txt:
```diff
@@ -49,7 +50,6 @@ set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_DEBUG} -DDEBUG_IS_ON=1")
 
 # Add Gcov flags
 set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -g -O0 -fprofile-arcs -ftest-coverage")
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -O0 -fprofile-arcs -ftest-coverage")
```
Why do we want to remove the C++ flags here?

amarpad replied:
sorry must have been a rebase screw up. Let me double check

amarpad commented:
> So is the broad idea to do the following?
> 
> * put all src files into [directory/src/](url)
> * put any header files that need to be used somewhere else in `directory/includes/`

@themarwhal moving headers to includes helps us add a directive like target_include_directories(includes) that then becomes a transitive dependency for any cmake project that imports the target. This allows us to not hardcode the paths of include directories and pass it between independent cmake projects.
The src pattern is just to introduce a superbuild. CMake doesn't have a good way of capturing dependencies between independent projects and so I created a superbuild for cmake which just delegates to the sub builds and then marks them dependent on each other. This way all builds that depend on common are actually depending on common explicitly within cmake as opposed to today where we assume common got built magically

Lmk if it makes sense.

Commit:
Support build updates to support new paths

Trivial update to point at the right directories

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

amarpad commented:
@electronjoe 
https://github.com/magma/magma/pull/6423/checks?check_run_id=2486819711
Looks like the issue is an existing file that is failing lints for the python side. This change is pretty large, lmk if I should address lint issues as well in this change or we can ignore it for now. Happy to go either way.

rdefosse commented:
@amarpad 

Looks better. Did you change the location of the sctpd executable or the name?

```bash
Step 62/79 : COPY --from=magma-mme-builder $C_BUILD/sctpd/sctpd .
COPY failed: stat /docker/overlay2/e3dda8ddb4d269281bdf64c7e63d10422767ef2b8cdc953836a16cd81686f935/merged/build/c/sctpd/sctpd: no such file or directory
MAGMA-OAI-MME DOCKER IMAGE BUILD: KO
```

The build part went OK:

```bash
[1/14] Running C++ protobuf compiler on /magma/lte/protos/sctpd.proto
[2/14] Running GRPC protobuf compiler on /magma/lte/protos/sctpd.proto
[3/14] Building CXX object CMakeFiles/SCTPD_LIB.dir/sctp_desc.cpp.o
[4/14] Building CXX object CMakeFiles/SCTPD_LIB.dir/lte/protos/sctpd.pb.cc.o
[5/14] Building CXX object CMakeFiles/SCTPD_LIB.dir/sctp_assoc.cpp.o
[6/14] Building CXX object CMakeFiles/SCTPD_LIB.dir/util.cpp.o
[7/14] Building CXX object CMakeFiles/SCTPD_LIB.dir/sctpd_downlink_impl.cpp.o
[8/14] Building CXX object CMakeFiles/SCTPD_LIB.dir/sctpd_event_handler.cpp.o
[9/14] Building CXX object CMakeFiles/SCTPD_LIB.dir/sctpd_uplink_client.cpp.o
[10/14] Building CXX object CMakeFiles/sctpd.dir/sctpd.cpp.o
[11/14] Building CXX object CMakeFiles/SCTPD_LIB.dir/sctp_connection.cpp.o
[12/14] Building CXX object CMakeFiles/SCTPD_LIB.dir/lte/protos/sctpd.grpc.pb.cc.o
[13/14] Linking CXX static library libSCTPD_LIB.a
[14/14] Linking CXX executable sctpd
[15/16] No install step for 'Sctpd'
[16/16] Completed 'Sctpd'
```

themarwhal commented:
> @electronjoe
> [#6423 (checks)](https://github.com/magma/magma/pull/6423/checks?check_run_id=2486819711)
> Looks like the issue is an existing file that is failing lints for the python side. This change is pretty large, lmk if I should address lint issues as well in this change or we can ignore it for now. Happy to go either way.

I think we can ignore the linter check failure for this PR. (This check will fail usually because a lot of our codebase doesn't comply with the linter. I only ask PR authors to fix them up when the lines they touched produce a liter warning. In that case we would see inline annotations in the file itself.)
Like this:
<img width="754" alt="Screen Shot 2021-05-03 at 7 54 18 AM" src="https://user-images.githubusercontent.com/37634144/116878296-c4f1b480-abe4-11eb-897f-eee92cb43f7a.png">

amarpad commented on ci-scripts/docker/Dockerfile.mme.ci.rhel8:
```diff
@@ -89,8 +89,8 @@ RUN ldconfig -v
-COPY --from=magma-mme-builder $C_BUILD/oai/oai_mme/mme oai_mme
-COPY --from=magma-mme-builder $C_BUILD/sctpd/sctpd .
+COPY --from=magma-mme-builder $C_BUILD/core/oai/oai_mme/mme oai_mme
+COPY --from=magma-mme-builder $C_BUILD/sctpd/src/sctpd .
```
@rdefosse is this the line that looks for sctpd? Is there a way to build this locally so I can test locall?

rdefosse commented on ci-scripts/docker/Dockerfile.mme.ci.ubuntu18:
```diff
@@ -97,7 +97,7 @@ RUN ldconfig
 WORKDIR /magma-mme/bin
-COPY --from=magma-mme-builder $C_BUILD/oai/oai_mme/mme oai_mme
+COPY --from=magma-mme-builder $C_BUILD/core/oai/oai_mme/mme oai_mme
 COPY --from=magma-mme-builder $C_BUILD/sctpd/sctpd .
```
@amarpad 

Here --> you still have the old location for `sctpd` if I am correct?

rdefosse commented on ci-scripts/docker/Dockerfile.mme.ci.ubuntu18:
```diff
@@ -117,7 +117,7 @@ COPY --from=magma-mme-builder $MAGMA_ROOT/lte/gateway/configs/service_registry.y
 # Scripts to re-generate certificates
 WORKDIR /magma-mme/scripts
-COPY --from=magma-mme-builder $MAGMA_ROOT/lte/gateway/c/oai/test/check_mme_s6a_certificate .
+COPY --from=magma-mme-builder $MAGMA_ROOT/lte/gateway/c/core/oai/test/check_mme_s6a_certificate .
```
Just checking if the location has really changed!

rdefosse commented on lte/gateway/docker/mme/Dockerfile.rhel8:
```diff
@@ -416,7 +416,7 @@ RUN yum install -y python3-pip && \
 WORKDIR /magma-mme/bin
-RUN cp $C_BUILD/oai/oai_mme/mme oai_mme
+RUN cp $C_BUILD/core/oai/oai_mme/mme oai_mme
 RUN cp $C_BUILD/sctpd/sctpd .
```
Is the `sctpd` location correct?

amarpad replied:
Nope :D. It needs to be under src. I think I don't know how to build this locally? Will do that and then report back. Thanks for the instructrions.

rdefosse commented on lte/gateway/docker/mme/Dockerfile.rhel8:
```diff
@@ -435,7 +435,7 @@ RUN cp $MAGMA_ROOT/lte/gateway/configs/redis.yml .
 
 WORKDIR /magma-mme/scripts
-RUN cp $MAGMA_ROOT/lte/gateway/c/oai/test/check_mme_s6a_certificate .
+RUN cp $MAGMA_ROOT/lte/gateway/c/core/oai/test/check_mme_s6a_certificate .
```
Is the location being changed?

amarpad replied:
Yup.

rdefosse commented on lte/gateway/docker/mme/Dockerfile.rhel8:
```diff
@@ -509,7 +509,7 @@ RUN ldconfig -v
 WORKDIR /magma-mme/bin
-COPY --from=magma-mme-builder $C_BUILD/oai/oai_mme/mme oai_mme
+COPY --from=magma-mme-builder $C_BUILD/core/oai/oai_mme/mme oai_mme
 COPY --from=magma-mme-builder $C_BUILD/sctpd/sctpd .
```
Is `sctpd` location correct?

amarpad replied:
Nope :(

rdefosse commented on lte/gateway/docker/mme/Dockerfile.rhel8:
```diff
@@ -531,7 +531,7 @@ COPY --from=magma-mme-builder $MAGMA_ROOT/lte/gateway/configs/service_registry.y
 # Adding means to re-generate certificates
 WORKDIR /magma-mme/scripts
-COPY --from=magma-mme-builder $MAGMA_ROOT/lte/gateway/c/oai/test/check_mme_s6a_certificate .
+COPY --from=magma-mme-builder $MAGMA_ROOT/lte/gateway/c/core/oai/test/check_mme_s6a_certificate .
```
Did the location change?

amarpad replied:
Yep

Commit:
Update session_manager test to use module find

Update to use a module to find gmock

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

electronjoe commented:
> @themarwhal moving headers to includes helps us add a directive like target_include_directories(includes) that then becomes a transitive dependency for any cmake project that imports the target. This allows us to not hardcode the paths of include directories and pass it between independent cmake projects.
> The src pattern is just to introduce a superbuild. CMake doesn't have a good way of capturing dependencies between independent projects and so I created a superbuild for cmake which just delegates to the sub builds and then marks them dependent on each other. This way all builds that depend on common are actually depending on common explicitly within cmake as opposed to today where we assume common got built magically

It's increasingly unclear I'll find time to contribute to our build tooling - so I'm inclined to ask few questions and let you refactor at will @amarpad .

I was aiming to get us moving away from the collection-of-cmake-projects paradigm, though I am not certain it's possible.  I consider a monorepo with poly-repo build semantics to be non ideal.  The Include directory use is one feature of non-mono-repoism, but not the only one. Possibly with Cmake this is the only path - I don't know.  I have some outstanding GH Issues proposing changes to explicitly remove the include directory style, and to give full include paths on import (from root of magma).

I'm frankly not informed enough in Cmake (yet) to really weigh in in this particular solution.  Is there best practice documentation that led you this direction?

amarpad commented:
> > @themarwhal moving headers to includes helps us add a directive like target_include_directories(includes) that then becomes a transitive dependency for any cmake project that imports the target. This allows us to not hardcode the paths of include directories and pass it between independent cmake projects.
> > The src pattern is just to introduce a superbuild. CMake doesn't have a good way of capturing dependencies between independent projects and so I created a superbuild for cmake which just delegates to the sub builds and then marks them dependent on each other. This way all builds that depend on common are actually depending on common explicitly within cmake as opposed to today where we assume common got built magically
> 
> It's increasingly unclear I'll find time to contribute to our build tooling - so I'm inclined to ask few questions and let you refactor at will @amarpad .
> 
> I was aiming to get us moving away from the collection-of-cmake-projects paradigm, though I am not certain it's possible. I consider a monorepo with poly-repo build semantics to be non ideal. The Include directory use is one feature of non-mono-repoism, but not the only one. Possibly with Cmake this is the only path - I don't know. I have some outstanding GH Issues proposing changes to explicitly remove the include directory style, and to give full include paths on import (from root of magma).
> 
> I'm frankly not informed enough in Cmake (yet) to really weigh in in this particular solution. Is there best practice documentation that led you this direction?

I ended up reading this: https://crascit.com/professional-cmake/, I am reasonably confident that this is the best approach with super builds.

I think the fundamental challenge here is do we want to treat libcommon as a separate library that services link against? If so what I am doing is the best practice. If not we should move common out of there and make a single CMake file that builds everything (with targets obviously) then all public headers go into lib/includes or something and everyone knows that they need to include this (this is probably closer to what you are suggesting). My understanding of why common was a separate library was that it was platform functionality and we didn't want to put it under lte. Maybe what you are saying is we just blow that part up fully and have a single C/C++ build tree as anyway we are already including 5G and WiFI in our builds. 
The current change that I am doing isn't changing the fact that libcommon is a separate library it is just making it so that cmake captures the dependencies and it is not sheer luck or explicitly captured in a wrapper Makefile at lte/gateway/magma
It all started with the need to make it easier for ITTI tasks to be buildable independently for that I need to be able to capture the dependencies in CMake and not in an external Makefile as it just doesn't work otherwise.

uri200 commented:
just adding it here so we have all the logs in one place. Those changes prevents us loading the project properly in intellij. We lose some basic features asa the debugger debugger. 

I have even tried to move `ExternalProject_Add` from the lower level into sessiond cmake, but it didn't help.

The issue seems to be the project can't locate common libs when run by itself. 

![image](https://user-images.githubusercontent.com/16157139/117222276-ef4b9980-adbf-11eb-8d98-804d1af48346.png)


```
/usr/bin/cmake -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTS=true -G "CodeBlocks - Unix Makefiles" /tmp/tmp.4jyaYp85Tj
-- The C compiler identification is GNU 6.3.0
-- The CXX compiler identification is GNU 6.3.0
-- Check for working C compiler: /usr/lib/ccache/cc
-- Check for working C compiler: /usr/lib/ccache/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/lib/ccache/c++
-- Check for working CXX compiler: /usr/lib/ccache/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Found CLANG_FORMAT: /usr/bin/clang-format
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found Protobuf: /usr/lib/x86_64-linux-gnu/libprotobuf.so;-lpthread (found version "3.0.0") 
Adding code coverage build and linker flags for BUILD_TESTS
Build type is Debug
-- Found Protobuf: /usr/lib/x86_64-linux-gnu/libprotobuf.so;-lpthread;-lpthread (found version "3.0.0") 
Proto_srcs are /tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/orc8r/protos/common.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/orc8r/protos/directoryd.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/lte/protos/session_manager.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/lte/protos/subscriberdb.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/lte/protos/policydb.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/lte/protos/pipelined.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/lte/protos/spgw_service.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/lte/protos/mconfig/mconfigs.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/feg/gateway/services/aaa/protos/accounting.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/feg/gateway/services/aaa/protos/context.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/orc8r/protos/directoryd.grpc.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/lte/protos/session_manager.grpc.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/lte/protos/pipelined.grpc.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/lte/protos/spgw_service.grpc.pb.cc;/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/feg/gateway/services/aaa/protos/accounting.grpc.pb.cc
CMake Error at CMakeLists.txt:95 (find_package):
  By not providing "FindASYNC_GRPC.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "ASYNC_GRPC", but CMake did not find one.

  Could not find a package configuration file provided by "ASYNC_GRPC" with
  any of the following names:

    ASYNC_GRPCConfig.cmake
    async_grpc-config.cmake

  Add the installation prefix of "ASYNC_GRPC" to CMAKE_PREFIX_PATH or set
  "ASYNC_GRPC_DIR" to a directory containing one of the above files.  If
  "ASYNC_GRPC" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring incomplete, errors occurred!
See also "/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/CMakeFiles/CMakeOutput.log".
See also "/tmp/tmp.4jyaYp85Tj/cmake-build-debug-remote-host-2222/CMakeFiles/CMakeError.log".

[Failed to reload]
```

Commit:
Update CI script path to new location

Signed-off-by: Amar Padmanabhan <amarpadmanabhan@fb.com>

electronjoe commented:
> just adding it here so we have all the logs in one place. Those changes prevents us loading the project properly in intellij. We lose some basic features asa the debugger debugger.

Yeah, would be nice not to regress your abilities.

- What intellij is this? CLion or IDEA?
- Do you know what cmake version is getting used by this setup? We should compare with what Amar is using.
- What CMake version are you running Amar? Are you doing test dev in the Vagrant Ubuntu 20.04 vm?

uri200 commented:
> > just adding it here so we have all the logs in one place. Those changes prevents us loading the project properly in intellij. We lose some basic features asa the debugger debugger.
> 
> Yeah, would be nice not to regress your abilities.
> 
> * What intellij is this? CLion or IDEA?
> * Do you know what cmake version is getting used by this setup? We should compare with what Amar is using.
> * What CMake version are you running Amar? Are you doing test dev in the Vagrant Ubuntu 20.04 vm?

Version of CLion is 2021.1.1 (latest) But I have been using the previous versions without any problem with this project.

CLion allows Vagrant integration, so Cmake and compiler is from vagrant itself. Whatever it is installed in Vagrant, we use it (right now cmake is 3.7.2m GDB 7.12)

Right now I am using the default VM we have on master which is debian. No Ubunty

Note the request I am making is not about CLion, but about the **ability to run the IDE debugger**. If this project runs on VScode the same way it was running with CLion then we can switch.

This is the debugger running 

![image](https://user-images.githubusercontent.com/16157139/117333699-0afd8100-ae4e-11eb-978d-50c2a54e2e1e.png)

uri200 commented:
What should be the instructions to try to build this so we can run debugger on sessiond?

What I tried so far is to load sessiond project, build it. Then load src, but during build I still see the common libraries are not found. I even try to open the project in the same window so previous project built objects would remain

```
Proto_srcs are /tmp/tmp.QRTkT15Vda/cmake-build-debug/orc8r/protos/common.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/orc8r/protos/directoryd.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/lte/protos/session_manager.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/lte/protos/subscriberdb.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/lte/protos/policydb.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/lte/protos/pipelined.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/lte/protos/spgw_service.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/lte/protos/mconfig/mconfigs.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/feg/gateway/services/aaa/protos/accounting.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/feg/gateway/services/aaa/protos/context.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/orc8r/protos/directoryd.grpc.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/lte/protos/session_manager.grpc.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/lte/protos/pipelined.grpc.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/lte/protos/spgw_service.grpc.pb.cc;/tmp/tmp.QRTkT15Vda/cmake-build-debug/feg/gateway/services/aaa/protos/accounting.grpc.pb.cc
CMake Error at CMakeLists.txt:95 (find_package):
  By not providing "FindASYNC_GRPC.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "ASYNC_GRPC", but CMake did not find one.

  Could not find a package configuration file provided by "ASYNC_GRPC" with
  any of the following names:
```

amarpad commented:
All the changes other than the sessiond one is on master, so closing
